### ISSUE TITLE
List doc output too dark on Windows

### ISSUE THREAD

**Issue by runeimp** [casey/just#1294]
The beautiful blue doc comments in the `--list` output on macOS is a primary blue on Windows in both PowerShell and CMD. It's hard to read against the black background in CMD and next to invisible on the default blue background of PowerShell. Could have default to a softer/brighter blue for the doc comments on Windows? 😀

**Comment by casey** [comment]
Definitely! Could you post a screenshot?

**Comment by flw-cn** [comment]
This should be related to the color scheme theme of the terminal. I'm using iTerm2 and I don't think it's a "beautiful blue", I think it's very hard to read and can easily damage the eyes.

Here's the effect I see here, I'm using iTerm2, and the standard xterm theme:

<img width="1061" alt="image" src="https://user-images.githubusercontent.com/5546718/181525764-fad35195-e502-4180-80f8-69e77b66f22c.png">

**Comment by flw-cn** [comment]
If possible, I would like to be able to somehow configure the colors used by just, like LSCOLOR does. Or at least replace blue(34) with bright blue(94).

**Comment by runeimp** [comment]
I guess the beautiful blue is only in KiTTY on macOS
![macOS - KiTTY - Screen Shot 2022-07-28 at 10 17 21](https://user-images.githubusercontent.com/1418110/181600185-b11be6b5-fb28-4c12-be5b-4f1bf4000f76.png)

Here is Terminal on macOS
![macOS - Terminal - Screen Shot 2022-07-28 at 10 19 20](https://user-images.githubusercontent.com/1418110/181600337-cbef55d3-f1a0-46b9-a579-8f1a07999084.png)

Windows Command Prompt
![Windows - Command Prompt - Screen Shot 2022-07-28 at 10 17 59](https://user-images.githubusercontent.com/1418110/181600408-b4b2f257-d101-4cf7-a75b-2611cf6b3e21.png)

Windows PowerShell
![Windows - PowerShell - Screen Shot 2022-07-28 at 10 21 16](https://user-images.githubusercontent.com/1418110/181600440-72e4e552-ca41-4e5a-95b3-b47e3e503d29.png)

In all cases of macOS it is Big Sur and
in all cases of Windows it is Windows 10

**Comment by casey** [comment]
Nice, great screenshots, they definitely make the problem clear! I won't have time to hack on this until Saturday, but if someone wants to take a crack at it before then, I'm on board with using bright blue everywhere we're using normal blue.

Also, something else that would be super helpful, would be screenshots of what different colored text looks in those terminals, so we know what colors are legible and which to avoid. Something like this: https://askubuntu.com/questions/27314/script-to-display-all-terminal-colors

**Comment by runeimp** [comment]
Ideally the app would check the current background color and then do some calculation or check a table for using optimal colors against that background. As background colors are always configurable and can be as wildly different from the defaults as people are in general.

For instance the screenshots are all with app and system defaults. but I often change the defaults for most anything to Ocean, Mariana, Monokai, or a similar color theme. I just haven't gotten around to it yet since I reinstalled macOS on this system earlier this year.

**Comment by casey** [comment]
Is it possible to check the current background color? That would be cool, but probably too much work. I think blue is just a really bad choice. I assume that bright colors should show up well, regardless of the theme.

**Comment by runeimp** [comment]
In general yes. But can you use a library like [termbg](https://crates.io/crates/termbg) or similar to do the work for you? Or are you trying to avoid dependencies?

**Comment by casey** [comment]
We could, and I'm not opposed to adding dependencies, although termbg looks a little heavy. I'd say that for now we should get rid of the dark blue, which is definitely bad, and then see if anyone else complains 😄

**Comment by YDX-2147483647** [comment]
I think this is about Command Prompt / Windows PowerShell _terminal_, rather than _Windows_ as the title says.

![](https://user-images.githubusercontent.com/73375426/212448627-9d260bfd-97fb-435f-83b5-045d88e033c9.png)

↑ Windows Terminal on Windows 10, whatever your shell is. (I've tried cmd, powershell, and pwsh). Even if texts are still unreadable in Windows Terminal, we can [remap the colors](https://learn.microsoft.com/en-us/windows/terminal/customize-settings/color-schemes#creating-your-own-color-scheme).

_Remark_: [Windows Terminal is now the default in Windows 11](https://devblogs.microsoft.com/commandline/windows-terminal-is-now-the-default-in-windows-11/), so I guess it won't be an issue in near future — maybe years, though.

---

Besides, other CLI tools are unhappy with it either.

![](https://user-images.githubusercontent.com/73375426/212448875-5f390b84-7743-46fd-8e1b-5c27a6840ea3.png)

↑ [Poetry](https://python-poetry.org/) with ANSI output in Windows PowerShell default color theme. I've selected “ing dependencies …” so you can see where they are.

---

If your prefer to stick to Windows PowerShell terminal, here are some links about changing the theme. (sorry I didn't try)

- [about_ANSI_Terminals](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_ansi_terminals?view=powershell-7.3)
- [Set-PSReadLineOption](https://learn.microsoft.com/en-us/powershell/module/psreadline/set-psreadlineoption?view=powershell-5.1)

**Comment by bestlem** [comment]
And and example from iTerm on mac
![Screenshot 2023-07-03 at 18 08 20](https://github.com/casey/just/assets/37385/9d0da31a-5d0f-4526-82f0-957855a7463a)

An easy way might be to have a colour set for dark background rather than fully configurable.

so 
just --color=dark

or possibly have en environment variable to choose theme - which to start could just be dark or light or nocolor

Bright blue is not the solution any blue - except cyan is unreadable on a black background.

**Comment by zachvalenta** [comment]
Not sure if everyone following this issue gets an email from Github based on the mention in the PR itself, so: I opened a [PR](https://github.com/casey/just/pull/3075) to allow config of all output elements from `just -l`.

Here's what I've got in my `zsh.rc` rn:
```sh
export JUST_LIST_RECIPE_COLOR='#fab387'   # peach
export JUST_LIST_DOC_COLOR='#f5c2e7'      # pink
export JUST_LIST_GROUP_COLOR='#a6e3a1'    # mint green
export JUST_LIST_ALIAS_COLOR='#94e2d5'    # teal
```

in an attempt at something like catppuccin:

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/c21ef9a5-b53c-4a22-8613-e2c0056b97cf" />

**Comment by casey** [comment]
I've been thinking about this a little bit, and I think that this is not actually really a just issue, and the fix is that users should configure their terminal colors. If the default blue color is not legible on the background color, this is definitely an issue with the terminal color scheme, and not something that just should try to workaround.

As far as I know, all the terminals on windows are configurable, so even if we did add the ability to configure this to just, it would be just as much work for users to configure just than it would be to configure the colors on their terminal, and that functionality already exists, and would fix blue not being visible for all programs, not just just.

**Comment by flw-cn** [comment]
@casey I strongly disagree with your viewpoint. There are so many CLI/TUI applications out there that support custom color schemes, including Just itself. If we follow your logic, the developers of these applications must be completely out of their minds.

However, I fully acknowledge that you have complete autonomy in this matter, and none of us can convince you otherwise.

So, I’m offering a workaround here for other users who, like me, are deeply affected by this issue.

You can write a simple wrapper like this—in fact, I’ve named it `just-list`—and then use `just-list` instead of `just --list`.

```
fish ❯ cat just-list
just --color always --list | sed 's/\x1b\[34m/\x1b\[32m/g'
```

Furthermore, I usually write this at the beginning of my Justfile:

```
fish ❯ cat Justfile
default: help

# Show helps
help:
    just-list
```

So whenever I type `just`, I’ll see the prompt:

<img width="1537" height="641" alt="Image" src="https://github.com/user-attachments/assets/18491973-91a2-4334-8690-ab5439b26e53" />

**Comment by casey** [comment]
@flw-cn Sorry if I'm missing something, but is it possible to change the blue color to something lighter which shows up better? A lot of programs offer general theming support, but that's usually for aesthetic reasons, and not to work-around problems with the terminal theme.
